sst/SimEngCoreWrapper.cc

```diff
@@ -0,0 +1,368 @@
+      // delimiting or escape behavior even with double quotes.
+      // e.g "arg1=1 arg2='"Hi"' arg3=2" will be parsed as
+      // std::vector<std::string>{arg1=1, arg2="Hi", arg3=2}
+      if (currChar == '\'' && escapeSingle) {
+        escapeSingle = 0;
+      }
+      // If a portion of the argument string starts with a double quote (") and
+      // we encounter another double quote, capture the substring enclosed by a
+      // valid set of double quotes into an argument without producing any
+      // delimiting or escape behavior even with single quotes.
+      // e.g "arg1=1 arg2="James' Car" arg3=2" will be parsed as
+      // std::vector<std::string>{arg1=1, arg2=James' Car, arg3=2}
+      else if (currChar == '\"' && escapeDouble) {
+        escapeDouble = 0;
+      } else {
+        str += currChar;
+      }
+    } else {
+      if (currChar == ' ') {
+        if (str != "") {
+          args.push_back(str);
+          str = "";
+        }
+      }
+      // Check for escape character ("), this signals the algorithm to capture
+      // any char inside a set of ("") without producing any delimiting or
+      // escape behavior.
+      else if (currChar == '\"') {
+        escapeDouble = 1;
+        // Check for escape character ('), this signals the algorithm to capture
+        // any char inside a set of ('') without producing any delimiting or
+        // escape behavior.
+      } else if (currChar == '\'') {
+        escapeSingle = 1;
+      } else {
+        str += currChar;
+      }
+    }
+  }
+  if (escapeSingle || escapeDouble) {
+    std::string err;
+    output_.verbose(CALL_INFO, 1, 0, R"(
+           Parsing failed: Invalid format - Please make sure all
+           characters/strings are escaped properly within a set single or 
+           double quotes. To escape quotes use (\\\) instead of (\).\n
+           )");
+    std::cerr << "Error occured at index " << index
+              << " of the argument string - substring: "
+              << "[ " << str << " ]" << std::endl;
+    std::exit(EXIT_FAILURE);
+  }
+  args.push_back(str);
+  return args;
+}
+
+void SimEngCoreWrapper::fabricateSimEngCore() {
+  output_.verbose(CALL_INFO, 1, 0, "Setting up SimEng Core\n");
+  if (simengConfigPath_ != "") {
+#ifdef SIMENG_ENABLE_SST_TESTS
+    if (assembleWithSource_) {
+      output_.verbose(CALL_INFO, 1, 0,
+                      "Assembling source instructions using LLVM\n");
+      Assembler assemble = Assembler(source_);
+      coreInstance_ = std::make_unique<simeng::CoreInstance>(
+          assemble.getAssembledSource(), assemble.getAssembledSourceSize(),
+          simengConfigPath_);
+    } else {
+      coreInstance_ = std::make_unique<simeng::CoreInstance>(
+          simengConfigPath_, executablePath_, executableArgs_);
+    }
+#else
+    coreInstance_ = std::make_unique<simeng::CoreInstance>(
+        simengConfigPath_, executablePath_, executableArgs_);
+#endif
```

Comment:
Having seen this [comment](https://github.com/UoB-HPC/SimEng/pull/271#discussion_r1016632653), I understand the need for the `ifdef`.  What is your opinion on just having the `ifdef` around line 251 rather than the whole clause? the associated `else` could exit with an error message saying "Build wth test suite on" (or words to that effect) and would make this code sequence easier to read in my opinion

Reply:
Done.